custom/icds_reports/ucr/expressions.py

```diff
@@ -288,10 +293,36 @@ class GetAppVersion(JsonObject):
     def configure(self, app_version_string):
         self._app_version_string = app_version_string
 
-    def __call__(self, item, context=None):
+    def get_version_from_app_object(self, item):
+        domain_name = item.get('domain')
+        form_accessor = FormAccessors(domain_name)
+        form_id = item.get('form').get('meta').get('instanceID')
+        form = form_accessor.get_form(form_id)
+
+        app = get_app(domain_name, form.app_id)
+        if isinstance(app, LinkedApplication):
+            app_version = app.upstream_version
+        else:
+            app_version = app.version
+
+        return app_version
+
+    def get_version_from_appverision_string(self, item, context):
         app_version_string = self._app_version_string(item, context)
         return get_version_from_appversion_text(app_version_string)
 
+    def __call__(self, item, context=None):
+        if settings.SERVER_ENVIRONMENT == 'india':
+            try:
+                app_version = self.get_version_from_app_object(item)
+            except Http404:
+                app_version = self.get_version_from_appverision_string(item, context)
```

Comment:
If getting the app fails and it's a linked app, this version is likely incorrect. Would it be better to return a blank value instead, or to let this fail instead of capturing the error here?

Reply:
Hey, My main motive for adding this is to allow the test cases to pass locally because UCR tests have the forms not necessarily app. 
And In case it fails to get the app from `app = get_app(domain_name, form.app_id)` it means the app does not exist (linked app even). In that case, I think it is safer to fall back to the original method which is to get the app version from the string in the form. Feel free to disagree, happy to discuss more.

Reply:
That makes sense, thanks for explaining.